### PR TITLE
relay: fix invalid flavor after restart dm-master

### LINKED ISSUE #993: invalid flavor after restart master
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
[jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6481/pipeline/75/)

***reproduce***:
start a task with one master and one worker, enable relay
```bash 
    while true; do
        ps aux | grep dm-master |awk '{print $2}'|xargs kill || true
        check_port_offline $MASTER_PORT1 20
        sleep 20
        run_dm_master $WORK_DIR/master $MASTER_PORT $cur/conf/dm-master.toml
        check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT

        run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
            "query-status test" \
            "\"result\": true" 2
    done
```
2. What did you expect to see?
task restart
3. What did you see instead?
WXWorkCapture_15992154724958

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
        master
        ```

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
close #993 

### What is changed and how it works?
copy source config to subtask task before `getMinPosInAllSubTasks`

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
 use shell script in #993 , currently dmctl_basic integration_test also cover it.

### CODE REVIEW

File: dm/worker/worker.go [hunk 1]
```diff
@@ -214,7 +214,7 @@ func (w *Worker) StartSubTask(cfg *config.SubTaskConfig) {
 	defer w.Unlock()
 
 	// copy some config item from dm-worker's config
-	w.copyConfigFromWorker(cfg)
+	copyConfigFromSource(cfg, w.cfg)
```
Comment (csuzhangxc) [inline dm/worker/worker.go:217]: Does this still need?

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM

File: dm/worker/worker.go [hunk 2]
```diff
@@ -796,18 +796,18 @@ func (w *Worker) OperateSchema(ctx context.Context, req *pb.OperateWorkerSchemaR
...
+	cfg.ServerID = sourceCfg.ServerID
+	cfg.RelayDir = sourceCfg.RelayDir
+	cfg.EnableGTID = sourceCfg.EnableGTID
+	cfg.UseRelay = sourceCfg.EnableRelay
 
 	// we can remove this from SubTaskConfig later, because syncer will always read from relay
-	cfg.AutoFixGTID = w.cfg.AutoFixGTID
+	cfg.AutoFixGTID = sourceCfg.AutoFixGTID
 
 	// log config items, mydumper unit use it
```
Comment (lance6716) [inline dm/worker/worker.go:812]: empty comment

File: dm/worker/worker.go [hunk 3]
```diff
@@ -214,7 +214,7 @@ func (w *Worker) StartSubTask(cfg *config.SubTaskConfig) {
 	defer w.Unlock()
 
 	// copy some config item from dm-worker's config
```
Comment (lance6716) [inline dm/worker/worker.go:216]:
```suggestion
	// copy some config item from dm-worker's source config
```

File: dm/worker/worker.go (hunk 1 above)
Comment (GMHDBJD) [reply dm/worker/worker.go:217]: Others such as `worker server` also call `StartSubTask` and use origin subtaskCfgs

**Review by lance6716 — COMMENTED** [review]
rest LGTM

File: dm/worker/worker.go (hunk 2 above)
Comment (GMHDBJD) [reply dm/worker/worker.go]: remove in 3e07cdd

**Review by csuzhangxc — COMMENTED** [review]
LGTM

**Comment by lance6716** [comment]
LGTM

**Review by lance6716 — APPROVED** [review]

**Pull request merged**